src/sinks/amqp.rs

```diff
@@ -0,0 +1,615 @@
+use crate::{
```

Comment:
I can see you borrowed the `kafka` sink implementation for this, which definitely makes sense!; however, I think that sink's implementation actually needs to be revisited because it can allow out-of-order publishes (https://github.com/timberio/vector/issues/7277), which this new sink also suffers from.

What would you think of simplifying the implementation here to publish messages one at a time, polling the future until it is sent? That will result in lower throughput, but is a simpler initial approach. We can revisit this later to add concurrency.

Reply:
@jszwedko I've updated the implementation which should preserve publishing order. Can you take a look again?

Reply:
Thanks! I'm reading what you have here as having swapped the futures pool for a `VecDeque`; is that accurate? I think that would work in that the futures are being polled in the order they are pushed onto the `VecDeque`. However, given this essentially does serial publishing (which was my suggestion), I think this code could be simplified further by simply only sending one message at a time and polling until it is complete. I believe that would be equivalent to what you have here and cut down on some of the complexity of managing multiple futures as you have here.

What do you think?